src/runners/rfq_maker_balance_cache_runner.ts

```diff
@@ -0,0 +1,169 @@
+
+// tslint:disable-next-line:custom-no-magic-numbers
+const DELAY_WHEN_NEW_BLOCK_FOUND = ONE_SECOND_MS * 5;
+const DELAY_WHEN_NEW_BLOCK_NOT_FOUND = ONE_SECOND_MS;
+// The eth_call will run out of gas if there are too many balance calls at once
+const MAX_BALANCE_CHECKS_PER_CALL = 1000;
+
+// Metric collection related fields
+const LATEST_BLOCK_PROCESSED_GAUGE = new Gauge({
+    name: 'rfqtw_latest_block_processed',
+    help: 'Latest block processed by the RFQ worker process',
+    labelNames: ['workerId'],
+});
+
+process.on('uncaughtException', err => {
+    logger.error(err);
+    process.exit(1);
+});
+
+process.on('unhandledRejection', err => {
+    if (err) {
+        logger.error(err);
+    }
+});
+
+interface BalancesCallInput {
+    addresses: string[];
+    tokens: string[];
+}
+
+if (require.main === module) {
+    (async () => {
+        logger.info('running RFQ balance cache runner');
+        const ethersProvider = new ethers.providers.JsonRpcProvider(
+            defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl,
+        );
+        const connection = await getDBConnectionAsync();
+        const balanceCheckerContractInterface = await _getBalanceCheckerContractInterfaceAsync(ethersProvider);
+
+        await runRfqBalanceCheckerAsync(ethersProvider, connection, balanceCheckerContractInterface);
+    })().catch(error => logger.error(error.stack));
+}
+
+async function runRfqBalanceCheckerAsync(
+    ethersProvider: ethers.providers.JsonRpcProvider,
+    connection: Connection,
+    balanceCheckerContractInterface: ethers.Contract,
+): Promise<void> {
+    const workerId = _.uniqueId('rfqw_');
+    let lastBlockSeen = -1;
+    while (true) {
+        const newBlock = await ethersProvider.getBlockNumber();
+        if (lastBlockSeen < newBlock) {
+            lastBlockSeen = newBlock;
+            LATEST_BLOCK_PROCESSED_GAUGE.labels(workerId).set(lastBlockSeen);
+            logUtils.log(
+                {
+                    block: lastBlockSeen,
+                    workerId,
+                },
+                'Found new block',
+            );
+
+            const makerTokens = await _getMakerTokensAsync(connection);
+            const balancesCallInput = _splitValues(makerTokens);
+
+            const updateTime = new Date();
+            const erc20Balances = await _getErc20BalancesAsync(balanceCheckerContractInterface, balancesCallInput);
+
+            await _updateErc20BalancesAsync(balancesCallInput, erc20Balances, connection, updateTime);
+
+            await delay(DELAY_WHEN_NEW_BLOCK_FOUND);
+        } else {
+            await delay(DELAY_WHEN_NEW_BLOCK_NOT_FOUND);
+        }
+    }
+}
+
+// NOTE: this only returns a partial entity class, just token address and maker address
+// Cache the query results to reduce reads from the DB
+let MAKER_TOKEN_CACHE: ResultCache<MakerBalanceChainCacheEntity[]>;
+const _getMakerTokensAsync = async (connection: Connection) => {
```

Comment:
Just to make it clearer, would you mind being explicit about the return type?

Reply:
returns the entity, add type